sdcm/utils/k8s.py

```diff
@@ -609,6 +614,251 @@ def stop(self, timeout=None):
+    def _read_stream(self) -> Iterator[str]:
+        while True:
+            if not self.watcher or self.watcher.closed:
+                self._open_stream()
+            line = self.watcher.readline()
+            yield line.decode("utf-8") if isinstance(line, bytes) else line
+
+    def _process_line(self, line: str) -> None:  # pylint: disable=too-many-branches,inconsistent-return-statements
+        # NOTE: line is expected to have following structure:
+        # {"type": "ADDED",
+        #  "object": {
+        #      "kind": "Endpoints",
+        #      "apiVersion": "v1",
+        #      "metadata": {
+        #          "name":"sct-cluster-dc-1-kind-0",
+        #          "namespace": "scylla",
+        #          ...
+        #      },
+        #      "subsets": [{
+        #          "addresses": [
+        #              {"ip": "10.244.218.195", ... },
+        #              ...
+        #          ],
+        #          "ports": [...]
+        #      }]
+        #  }
+        # }
+        data = {}
+        try:
+            LOGGER.debug("Processing following line: %s", line)
+            data = yaml.safe_load(line) or {}
+            metadata = data.get('object', {}).get('metadata', {})
+            namespace = metadata.get('namespace')
+            if not namespace:
+                raise KeyError("Cannot find 'namespace' in the line: %s" % line)
+            if namespace not in self.mapper_dict:
+                self.mapper_dict[namespace] = {}
+            # NOTE: following condition allows to skip processing of all the
+            #       load-balancer/headless endpoints that do not refer to any Scylla pod.
+            labels = metadata.get('labels', {})
+            if not all((key in labels.keys() for key in self.SCYLLA_PODS_EXPECTED_LABEL_KEYS)):
+                return None
+
+            if name := metadata.get('name'):
+                if name not in self.mapper_dict[namespace]:
+                    self.mapper_dict[namespace][name] = {}
+                if 'current_ip' not in self.mapper_dict[namespace][name]:
+                    self.mapper_dict[namespace][name]['current_ip'] = None
+                if 'old_ips' not in self.mapper_dict[namespace][name]:
+                    self.mapper_dict[namespace][name]['old_ips'] = []
+            for subset in data.get('object', {}).get('subsets', []):
+                for address in subset.get('addresses', []):
+                    current_ip_address = address['ip']
+                    if current_ip_address == self.mapper_dict[namespace][name]['current_ip']:
+                        continue
+                    old_ip_candidate = self.mapper_dict[namespace][name]['current_ip']
+                    self.mapper_dict[namespace][name]['current_ip'] = current_ip_address
+                    if not old_ip_candidate:
+                        break
+                    self.mapper_dict[namespace][name]['old_ips'].append(old_ip_candidate)
+                    LOGGER.info(
```

Comment:
let's leave this info

Reply:
It has no comment above to do something with it. So, it is intended to stay as is.